### ISSUE TITLE
feat: Implement training loop to overfit one batch

### ISSUE THREAD

**Issue by muuki2** [muuki2/makemore_clone#10]
Train the model on a single mini-batch until perfect accuracy is reached (debug sanity check).
